lib/semantics/check-return.cc

```diff
@@ -33,18 +30,17 @@ void ReturnStmtChecker::Leave(const parser::ReturnStmt &returnStmt) {
   // C1575 The scalar-int-expr is allowed only in the inclusive scope of a
   // subroutine subprogram.
   const auto &scope{context_.FindScope(context_.location().value())};
-  const auto *subprogramScope{FindContainingSubprogram(scope)};
-  if (!subprogramScope) {
-    context_.Say(
-        "RETURN must in the inclusive scope of a SUBPROGRAM"_err_en_US);
-    return;
-  }
-  if (returnStmt.v && subprogramScope->kind() == Scope::Kind::Subprogram) {
-    if (IsFunction(*subprogramScope->GetSymbol())) {
-      context_.Say(
-          "RETURN with expression is only allowed in SUBROUTINE subprogram"_err_en_US);
+  if (const auto *subprogramScope{FindContainingSubprogram(scope)}) {
```

Comment:
Which Block Data clause is this?

Reply:
It is not a `BLOCK DATA` clause.  It is cleaning up some code that I noticed could be improved after implementing changes to `FindProgramUnitContaining()` and searching for all of the sites in the code that should use it.  The `FindContainingSubprogram()` function in `lib/semantics/check-return.cc` could be written in terms of `FindProgramUnitContaining()` and reduce its tight coupling with internals of the scope tree, so I did so.

Reply:
Ok